doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1522,144 +1508,108 @@ Managing the local context
+   removes this constraint.  This lets you avoid specifying the asserted statement
+   completely before starting to prove it.
 
-.. tacv:: pose proof @term {? as @simple_intropattern}
-   :name: pose proof
+.. tacn:: pose proof @term {? @as_ipat }
+          pose proof ( @ident := @term )
+   :name: pose proof; _
 
-   This tactic behaves like :n:`assert @type {? as @simple_intropattern} by exact @term`
+   The first form behaves like :n:`assert @type {? @as_ipat } by exact @term`
    where :token:`type` is the type of :token:`term`. In particular,
-   :n:`pose proof @term as @ident` behaves as :n:`assert (@ident := @term)`
+   :n:`pose proof @term as @ident` behaves like :n:`assert (@ident := @term)`
    and :n:`pose proof @term as @simple_intropattern` is the same as applying the
    :token:`simple_intropattern` to :token:`term`.
 
-.. tacv:: epose proof @term {? as @simple_intropattern}
-   :name: epose proof
-
-   While :tacn:`pose proof` expects that no existential variables are generated by
-   the tactic, :tacn:`epose proof` removes this constraint.
-
-.. tacv:: pose proof (@ident := @term)
-
-   This is an alternative syntax for :n:`assert (@ident := @term)` and
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
-
-.. tacv:: epose proof (@ident := @term)
-
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+   The second form is equivalent to :n:`assert (@ident := @term)` and
+   :n:`pose proof @term as @ident`, following the model of
+   :n:`pose (@ident := @term)` but dropping the value of :token:`ident`.
 
-.. tacv:: enough (@ident : @type)
-   :name: enough
+.. tacn:: epose proof @term {? @as_ipat }
+          epose proof ( @ident := @term )
+   :name: epose proof; _
 
-   This adds a new hypothesis of name :token:`ident` asserting :token:`type` to the
-   goal the tactic :tacn:`enough` is applied to. A new subgoal stating :token:`type` is
-   inserted after the initial goal rather than before it as :tacn:`assert` would do.
-
-.. tacv:: enough @type
-
-   This behaves like :n:`enough (@ident : @type)` with the name :token:`ident` of
-   the hypothesis generated by Coq.
-
-.. tacv:: enough @type as @simple_intropattern
-
-   This behaves like :n:`enough @type` using :token:`simple_intropattern` to name or
-   destruct the new hypothesis.
+   While :tacn:`pose proof` expects that no existential variables are generated by
+   the tactic, :tacn:`epose proof` removes this constraint.  The second form
+   is equivalent to an :n:`eassert (@ident := @term)`.
 
-.. tacv:: enough (@ident : @type) by @tactic
-          enough @type {? as @simple_intropattern } by @tactic
+.. tacn:: enough ( @ident : @term ) {? by @ltac_expr3 }
```

Comment:
```suggestion
.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
```

BTW, it might make sense to move `enough` and `cut` closer to `assert`.

Reply:
The only thing in between is pose proof and that's only about 3 inches of text.  I can move pose proof below cut.  Or should it go next to pose?

Reply:
Yeah, move `pose proof` below `cut`.